cmds/time_sos/service.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"os/exec"
+	"time"
+)
+
+type TimeService struct {
+	Date string
+	Time string
+}
+
+func getCurrentDate() string {
+	return time.Now().Format("2006-01-02")
+}
+
+func getCurrentTime() string {
+	return time.Now().Format("15:04")
+}
+
+func (ts *TimeService) Update() {
+	// Update sets the time service to the current system time
```

Comment:
oh, sorry, you need to follow the Go doc standard. 
So the comment starts on the line right before Update.

You need it for all these: AutoSetTime, NewTimeService, 
e.g.


// Ldd returns a list of all library dependencies for a
// set of files, suitable for feeding into (e.g.) a cpio
// program. If a file has no dependencies, that is not an
// error. The only possible error is if a file does not
// exist, or it says it has an interpreter but we can't read
// it, or we are not able to run its interpreter.
// It's not an error for a file to not be an ELF, as
// this function should be convenient and the list might
// include non-ELF executables (a.out format, scripts)
func Ldd(names []string) ([]*FileInfo, error) {